src/main/scala/com/databricks/spark/sql/perf/mllib/MLClassificationBenchmarkable.scala

```diff
@@ -0,0 +1,110 @@
+package com.databricks.spark.sql.perf.mllib
+
+import com.databricks.spark.sql.perf._
+import com.typesafe.scalalogging.slf4j.Logging
+import org.apache.spark.sql._
+
+import scala.collection.mutable.ArrayBuffer
+
+// TODO(tjh) rename, this is not only about classification
+class MLClassificationBenchmarkable(
+    extraParam: ExtraMLTestParameters,
+    commonParam: MLTestParameters,
+    test: BenchmarkAlgorithm,
+    sqlContext: SQLContext)
+  extends Benchmarkable with Serializable with Logging {
+
+  import MLClassificationBenchmarkable._
+
+  private var testData: DataFrame = null
+  private var trainingData: DataFrame = null
+  val param = ClassificationContext(commonParam, extraParam, sqlContext)
+
+  override val name = test.getClass.getCanonicalName
+
+  override val executionMode: ExecutionMode = ExecutionMode.SparkPerfResults
+
+  override def beforeBenchmark(): Unit = {
+    logger.info(s"$this beforeBenchmark")
+    try {
+      testData = test.testDataSet(param)
+      testData.cache()
+      testData.count()
+      trainingData = test.trainingDataSet(param)
+      trainingData.cache()
+      trainingData.count()
+    } catch {
+      case e: Throwable =>
+        println(s"$this error in beforeBenchmark: ${e.getStackTraceString}")
+        throw e
+    }
+  }
+
+  override def doBenchmark(
```

Comment:
Is there a need to make this public and not protected?  Same for other methods/vals.